cms/admin/permissionadmin.py

```diff
@@ -38,6 +38,27 @@ class PagePermissionInlineAdmin(TabularInline):
     extra = 0  # edit page load time boost
     show_with_view_permissions = False
 
+    def has_change_permission(self, request, obj=None):
```

Comment:
there's no pagepermission permission.
did you get a failing test?

Reply:
No, all test passes. This is needed only for django 21 but it works also on dj111 and d20. It's needed because of the changes that django 21 made regarding admin. https://docs.djangoproject.com/en/2.1/releases/2.1/#custom-admin-forms-need-to-take-the-view-only-case-into-account. And in code especially: https://github.com/olivierdalang/django/commit/63b62af169f6fc02182e35a058dcac78bdaff049#diff-3c42de3e53aba87b32c494f995a728dfR2024.

Reply:
https://github.com/django/django/blob/2.1b1/django/contrib/admin/options.py#L2060 this is causing the need for this code. In django 2.1 they do this and if user dont have sufficient permissions fields are removed. And it's causing test errors. It's somehow described here: https://docs.djangoproject.com/en/2.1/releases/2.1/#custom-admin-forms-need-to-take-the-view-only-case-into-account. Solution showed by django docs it's for very simple case, and its not working for us.

Reply:
@czpython `obj` here is always `Page` model, not `PagePermission` seems like django works like that in `ModelAdmin` in `changeform_view>>_changeform_view>>_create_formsets>>get_formsets_with_inlines>>get_inline_instances(self, request, obj=None)`